pkg/tfbridge/provider.go

```diff
@@ -726,6 +726,7 @@ func (p *Provider) Configure(ctx context.Context,
 func (p *Provider) Check(ctx context.Context, req *pulumirpc.CheckRequest) (*pulumirpc.CheckResponse, error) {
 	ctx = p.loggingContext(ctx, resource.URN(req.GetUrn()))
 	urn := resource.URN(req.GetUrn())
+	ctx = WithUrn(ctx, urn)
```

Comment:
I'm amenable to this but the question becomes why only in Check? Perhaps you could retrofit `p.loggingContext` to do this; it's already installed in all the methods pretty much.